modules/ROOT/examples/java/src/com/couchbase/code_snippets/Examples.java

```diff
@@ -139,132 +136,531 @@ public static void main (String [] args) throws CouchbaseLiteException, Interrup
+        mutableDoc.setString(Prop_Language, listLangs.get(RANDOM.nextInt(listLangs.size() -1)));
         database.save(mutableDoc);
 
-        Document document = database.getDocument(mutableDoc.getId());
-        // Log the document ID (generated by the database) and properties
-        outputMessage("Document ID :: " + document.getId());
-        outputMessage("Learning " + document.getString(Prop_Language));
+        Document newDoc = database.getDocument(mutableDoc.getId());
+        // Show the document ID (generated by the database) and properties
+        outputMessage("Document ID :: " + newDoc.getId());
+        outputMessage("Learning " + newDoc.getString(Prop_Language));
 
         // Create a query to fetch documents of type SDK.
         outputMessage("== Executing Query 1");
-        Query query = QueryBuilder.select(SelectResult.expression(Meta.id),
+        Query query = QueryBuilder.select(SelectResult.all())
+                .from(DataSource.database(database))
+                .where(Expression.property(Prop_Type).equalTo(Expression.string(searchStringType)));
+        ResultSet result = query.execute();
+        outputMessage(String.format("Query returned %d rows of type %s", result.allResults().size(), searchStringType));
+
+        // Create a query to fetch all documents.
+        outputMessage("== Executing Query 2");
+        Query queryAll = QueryBuilder.select(SelectResult.expression(Meta.id),
                 SelectResult.property(Prop_Language),
-                SelectResult.property(Prop_Version))
-                .from(database(database))
-                .where(Expression.property(Prop_Type).equalTo(Expression.string("SDK")));
-        ResultSet results = query.execute();
+                SelectResult.property(Prop_Version),
+                SelectResult.property(Prop_Type))
+                .from(DataSource.database(database));
+        ResultSet results = queryAll.execute();
         Result thisResult;
         String resultLine="";
         while( (thisResult=results.next())!=null) {
             NUMROWS++;
-            resultLine=String.format("%d ... Id: %s is learning: %s version: %.2f",
+            outputMessage(String.format("%d ... Id: %s is learning: %s version: %.2f type is %s",
                     NUMROWS,
-                    thisResult.getString("id"),
+                    thisResult.getString(Prop_Id),
                     thisResult.getString(Prop_Language),
-                    thisResult.getDouble(Prop_Version)
-                    );
-            outputMessage(resultLine);
-        }
-        outputMessage( String.format("Total rows returned by query = %d", NUMROWS));
+                    thisResult.getDouble(Prop_Version),
+                    thisResult.getString(Prop_Type)));
+        }        outputMessage( String.format("Total rows returned by query = %d", NUMROWS));
+
+        syncTotal = initiateReplication(database, SYNC_GATEWAY_URL, DB_USER, DB_PASS);
```

Comment:
In general, we would recommend users to start the continuous replicator only once, probably when the servlet or app got created. Otherwise, there will be multiple of the same replicator created and step to each other toe when syncing.  It may be fine to provide an example code like this but we probably need to make note about it so that users would know.

Reply:
No problem. I'll change this, and include a note to that effect.
For expediency, if I do a non-continuous replication in the app, does that avoid the issue here?